module/zfs/dsl_scan.c

```diff
@@ -762,6 +762,87 @@ dsl_scan_setup_sync(void *arg, dmu_tx_t *tx)
 	    *funcp, scn->scn_phys.scn_min_txg, scn->scn_phys.scn_max_txg);
 }
 
+static int
+dsl_rescrub_check(void *arg, dmu_tx_t *tx)
+{
+	dsl_scan_t *scn = dmu_tx_pool(tx)->dp_scan;
+ 	if (dsl_scan_is_running(scn))
+		return (SET_ERROR(EBUSY));
+	return (0);
+}
+
+static void
+dsl_rescrub_setup_sync(void *arg, dmu_tx_t *tx)
+{
+	uint64_t txg = tx->tx_txg;
+ 	dsl_scan_t *scn = dmu_tx_pool(tx)->dp_scan;
+ 	dsl_pool_t *dp = scn->scn_dp;
+	spa_t *spa = dp->dp_spa;
+ 	ASSERT(!dsl_scan_is_running(scn));
+	scn->scn_phys.scn_state = DSS_SCANNING;	
+
+ 	zfs_cmd_t zc = {"\0"};
+ 	uint64_t error_count = spa_get_errlog_size(spa);
+ 	if (error_count == 0){
+		scn->scn_phys.scn_state = DSS_FINISHED;
+		return;
+	}
+	zc.zc_nvlist_dst = (uintptr_t)kmem_zalloc(error_count * sizeof (zbookmark_phys_t), KM_SLEEP);
+	zc.zc_nvlist_dst_size = error_count;
+
+ 	size_t retrieved_error = (size_t)zc.zc_nvlist_dst_size;
+	int error = spa_get_errlog(spa, (void *)(uintptr_t)zc.zc_nvlist_dst,
+	    &retrieved_error);
```

Comment:
spa_get_errlog() is usually used to copy out to userland, which we aren't doing here.  Let's chat Monday about finding the right interface to use for this.